src/main/java/de/tum/in/www1/artemis/web/rest/lecture/AttachmentUnitResource.java

```diff
@@ -144,4 +152,52 @@ public ResponseEntity<AttachmentUnit> createAttachmentUnit(@PathVariable Long le
 
         return ResponseEntity.created(new URI("/api/attachment-units/" + savedAttachmentUnit.getId())).body(savedAttachmentUnit);
     }
+
+    /**
+     * POST lectures/:lectureId/attachment-units/split : creates new attachment units.
+     *
+     * @param lectureId            the id of the lecture to which the attachment units should be added
+     * @param lectureUnitSplitDTOs the units that should be created
+     * @param file                 the file to be splitted
+     * @return the ResponseEntity with status 200 (ok) and with body the newly created attachment units
+     * @throws IOException
```

Comment:
Should we really send the IOException to the client?

Reply:
I have to, because `attachmentUnitService.createAttachmentUnits(` throws an exception. How else could you handle?

Reply:
catch and log it on the server, and give the user a generic "could not create units" error.

I'd consider any IOException as an internal problem that the user does not need to know about.